python/foglamp/services/core/api/scheduler.py

```diff
@@ -339,6 +341,96 @@ def _extract_args(data, curr_value):
         raise web.HTTPNotFound(reason=str(ex))
 
 
+async def enable_schedule_with_name(request):
+    """ Enables the schedule for given schedule_name or schedule_id in request payload
+
+    curl -X PUT http://localhost:8081/foglamp/schedule/enable  -d '{"schedule_name": "a schedule name"}'
+
+    :param request: {"schedule_name": "sinusoid"} or {"schedule_id": "uuid of schedule"}
+    :return:
+    """
+    try:
+        data = await request.json()
+
+        sch_name = data.get('schedule_name', None)
+        sch_id = data.get('schedule_id', None)
+
+        if not sch_name and not sch_id:
+            raise web.HTTPBadRequest(reason='Schedule name or ID is required')
+
+        if sch_name and not sch_id:
+            storage_client = connect.get_storage_async()
+            payload = PayloadBuilder().SELECT("id").WHERE(['schedule_name', '=', sch_name]).payload()
+            result = await storage_client.query_tbl_with_payload('schedules', payload)
+
+            if int(result['count']):
+                sch_id = result['rows'][0]['id']
```

Comment:
Handle the case `count == 0`; otherwise it will throw 500 internal server error (i.e **sch_id** is None for this case)
```
{
    "error": {
        "message": "[TypeError] one of the hex, bytes, bytes_le, fields, or int arguments must be given"
    }
}
```

We should `raise ValueError('No schedule found')`

Reply:
when result['count'] value is 0, then it will be `if 0: `

Reply:
Then, when it reach to code line `assert uuid.UUID(None)`; it will throw TypeError